presto-main/src/main/java/com/facebook/presto/operator/TaskStats.java

```diff
@@ -192,113 +318,123 @@ public TaskStats(
             @JsonProperty("pipelines") List<PipelineStats> pipelines,
             @JsonProperty("runtimeStats") RuntimeStats runtimeStats)
     {
-        this.createTime = requireNonNull(createTime, "createTime is null");
-        this.firstStartTime = firstStartTime;
-        this.lastStartTime = lastStartTime;
-        this.lastEndTime = lastEndTime;
-        this.endTime = endTime;
-        this.elapsedTimeInNanos = elapsedTimeInNanos;
-        this.queuedTimeInNanos = queuedTimeInNanos;
-
-        checkArgument(totalDrivers >= 0, "totalDrivers is negative");
-        this.totalDrivers = totalDrivers;
-        checkArgument(queuedDrivers >= 0, "queuedDrivers is negative");
-        this.queuedDrivers = queuedDrivers;
-        checkArgument(queuedPartitionedDrivers >= 0, "queuedPartitionedDrivers is negative");
-        this.queuedPartitionedDrivers = queuedPartitionedDrivers;
-        checkArgument(queuedPartitionedSplitsWeight >= 0, "queuedPartitionedSplitsWeight must be non-negative");
-        this.queuedPartitionedSplitsWeight = queuedPartitionedSplitsWeight;
-
-        checkArgument(runningDrivers >= 0, "runningDrivers is negative");
-        this.runningDrivers = runningDrivers;
-        checkArgument(runningPartitionedDrivers >= 0, "runningPartitionedDrivers is negative");
-        this.runningPartitionedDrivers = runningPartitionedDrivers;
-        checkArgument(runningPartitionedSplitsWeight >= 0, "runningPartitionedSplitsWeight must be non-negative");
-        this.runningPartitionedSplitsWeight = runningPartitionedSplitsWeight;
+        this(Optional.ofNullable(createTime).map(DateTime::getMillis).orElse(0L),
+                Optional.ofNullable(firstStartTime).map(DateTime::getMillis).orElse(0L),
+                Optional.ofNullable(lastStartTime).map(DateTime::getMillis).orElse(0L),
+                Optional.ofNullable(lastEndTime).map(DateTime::getMillis).orElse(0L),
+                Optional.ofNullable(endTime).map(DateTime::getMillis).orElse(0L),
```

Comment:
In the interest of reducing object creation, we could just turn this into a utility method to avoid the `Optional` allocation throughout this PR.

```java
public static long getDateTimeMillis(@Nullable datetime) {
    if (datetime == null) {
         return 0L;
    }
    return datetime.getMillis();
}
```